google/gax/auth.py

```diff
@@ -31,19 +31,49 @@
 
 from __future__ import absolute_import
 
+import grpc
```

Comment:
There is a design restriction on `gax` not to import `grpc` in any module except `google.gax.grpc`. (The intention is that when we support alternative transports, we should only need to switch out a single module.) Let's move `secure_authorized_channel` to `grpc` for this reason. `gax` should also not expose any methods that are `grpc`-specific in its public interface, so let's make `secure_authorized_channel` "private".